src/deepsparse/schedulers/continuous_batching_scheduler.py

```diff
@@ -113,7 +113,7 @@ def submit(self, *args, operator: Operator, **kwargs) -> Future:
                 f"found {type(inputs)}"
             )
 
-        future = Future()
+        future = asyncio.Future() if kwargs.get("loop") else Future()
```

Comment:
Does this condition always hold when we run async? 🤔

Reply:
Yes

Reply:
Could you please put a comment on why these different futures are needed? I wasn't aware of this difference until reading https://stackoverflow.com/questions/29902908/what-is-the-difference-between-concurrent-futures-and-asyncio-futures